sdk/core/core-client-rest/src/urlHelpers.ts

```diff
@@ -57,3 +59,26 @@ export function buildRequestUrl(
       .replace(/([^:]\/)\/+/g, "$1")
   );
 }
+
+export function buildBaseUrl(baseUrl: string, options: RequestParameters): string {
+  if (!options.pathParameters) {
+    return baseUrl;
+  }
+  const pathParams = options.pathParameters;
+  for (const key of Object.keys(pathParams)) {
+    const param = pathParams[key] as any;
+    if (param === undefined || param === null) {
+      throw new Error(`Query parameters ${key} must not be undefined or null`);
+    }
+    if (!param.toString || typeof param.toString !== "function") {
+      throw new Error(`Query parameters must be able to be represented as string, ${key} can't`);
+    }
+    let value = param.toISOString !== undefined ? param.toISOString() : param.toString();
```

Comment:
Instead of checking for coercion ourselves, can't we let the native exception bubble up? Or do we want better logging? I guess we could always try/catch and rethrow?

```suggestion
    let value = param.toISOString !== undefined ? param.toISOString() : String(param);
```

Reply:
@xirzec hi, I tried to use the native exception in my local, then found out it was not clear enough to show the message, I will just keep the checking exception part. Let me know if you have other comments :)